app/src/main/java/net/gsantner/markor/activity/DocumentActivity.java

```diff
@@ -304,21 +332,16 @@ public void onBackPressed() {
     }
 
     public GsFragmentBase showFragment(GsFragmentBase fragment) {
-        GsFragmentBase currentTop = (GsFragmentBase) _fragManager.findFragmentById(R.id.document__placeholder_fragment);
 
-        if (currentTop == null || !currentTop.getFragmentTag().equals(fragment.getFragmentTag())) {
-            _fragManager.beginTransaction()
-                    //.addToBackStack(null)
-                    .replace(R.id.document__placeholder_fragment, fragment, fragment.getFragmentTag())
-                    .commit();
-        } else {
-            fragment = currentTop;
-        }
+        // Replace regardless of current state as document or paths may be different
+        _fragManager.beginTransaction()
+                .replace(R.id.document__placeholder_fragment, fragment, fragment.getTag())
+                .commit();
+
```

Comment:
Need to come up with a way to not do extra work if current document is same as requested document